File: module_practice_aop_annotation/src/main/java/com/alie/modulepracticeaopannotation/MainActivity.java
package com.alie.modulepracticeaopannotation;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    private MainPresenter presenter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        presenter = new MainPresenter();
        initView();
    }

    private void initView() {
        findViewById(R.id.btn1).setOnClickListener(v -> {
            presenter.onBtn1Click();
        });
        findViewById(R.id.btn2).setOnClickListener(v -> {
            presenter.onBtn2Click("西瓜");
        });
        findViewById(R.id.btn3).setOnClickListener(v -> {
            presenter.onBtn3Click("大西瓜", (short) 26);
        });
        findViewById(R.id.btn4).setOnClickListener(v -> {
            presenter.onBtn4Click(new Person("南瓜",56));
        });
    }
}

File: module_practice_anr/src/main/java/com/alie/modulepracticeanr/MainActivity.kt
package com.alie.modulepracticeanr

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mBtn1.setOnClickListener {
            // Let main thread sleep 20 sec and see what happens
            Thread.sleep(20 * 1000)
        }
    }
}
